src/index.ts

```diff
@@ -1,226 +1,180 @@
+    const sanitizedPath = path.startsWith('/') ? path.slice(1) : path;
+    const responsePromise = this.got(sanitizedPath, { ...gotOptions });
+
+    if (abortSignal) {
+      abortSignal &&
+        abortSignal.addEventListener('abort', () => {
+          responsePromise.cancel();
+        });
+    }
+    return responsePromise;
+  }
+
+  private async _requestJson<T>({ path, options }: { path: string; options: RequestOptions }): Promise<T> {
+    try {
+      const responsePromise = this._request(path, options);
+      const [response, json] = await Promise.all([responsePromise, responsePromise.json<T>()]);
+
+      this._logger?.info(
+        `${options.method} ${response.url} ${response.statusCode} ${new Date().getTime() - response.timings.start} ms`
+      );
+      this._logger?.debug('request-options', JSON.stringify(options).replace(/\\n/g, ''));
+      return json;
+    } catch (error) {
+      const { context, headers, method } = error.options;
+      const duration = (error.timings?.error || error.timings.end) - error.timings?.start;
+      const code = error.response?.statusCode || error.code;
+
+      if (error instanceof HTTPError || error instanceof CancelError) {
+        this._logger.error(
+          '\n' +
+            '--------------------------------------------------------------------\n' +
+            `${method} ${this._prefixUrl}${path} ${code || 'unkown statusCode'} (${duration ? duration : ' - '} ms)\n` +
+            `headers: ${JSON.stringify(headers)}\n` +
+            `request-options: ${JSON.stringify({ ...options, context }).replace(/\\n/g, '')}\n` +
+            `error:${error.message}\n` +
+            `stack:${error.stack}\n` +
+            '--------------------------------------------------------------------'
+        );
+      }
+      throw new RequestException({ message: error.message, statusCode: code, innerError: error, stack: error?.stack });
+    }
+  }
+
+  async get<T>(path: string, options?: RequestOptions): Promise<T> {
+    return this._requestJson({
+      path: path,
+      options: { ...options, method: 'GET' }
+    });
+  }
+  async post<T>(path: string, payload: Partial<T>, options?: RequestOptions): Promise<T> {
+    return this._requestJson({
+      path: path,
+      options: { ...options, method: 'POST', json: payload }
+    });
+  }
+  async put<T>(path: string, payload: Partial<T>, options?: RequestOptions): Promise<T> {
+    return this._requestJson({
+      path: path,
+      options: { ...options, method: 'PUT', json: payload }
+    });
+  }
+  async patch<T>(path: string, payload: Partial<T>, options?: RequestOptions): Promise<T> {
+    return this._requestJson({
+      path: path,
+      options: { ...options, method: 'PATCH', json: payload }
+    });
+  }
+  async delete(path: string, options?: RequestOptions): Promise<void> {
+    await this._requestJson({
+      path: path,
+      options: { ...options, method: 'DELETE' }
+    });
+    return void 0;
```

Comment:
Is this really needed? An undefined return in an async method _should_ match a `Promise<void>` I believe...

Reply:
Something I'm used to... but you're right.